### PR TITLE
Add test data interaction

### PR DESCRIPTION
**Pull request by trushkin-ilya**

### CODE REVIEW

**Comment by trushkin-ilya** [comment]
@grib0ed0v, requesting a review.

File: datasets.py [hunk 1]
```diff
@@ -41,13 +41,14 @@ def __getitem__(self, idx):
...
+    def __init__(self, protocol: int, dir: str = 'data/CASIA_SURF', mode: str = 'train', transform=None):
         self.dir = dir
-        file_name = f'4@{protocol}_{txt_metadata}.txt'
+        self.mode = mode
+        submode = {'train': 'train', 'dev': 'dev_ref', 'test': 'test_res'}[mode]
+        file_name = f'4@{protocol}_{submode}.txt'
         with open(os.path.join(dir, file_name), 'r') as file:
             lines = file.readlines()
-            if train:
+            if mode != 'test':
```
Comment (grib0ed0v) [inline datasets.py]: if self.mode would be better,

File: submit.py [hunk 2]
```diff
@@ -49,21 +84,22 @@
...
-                idx = np.arange(i * args.batch_size, (i + 1) * args.batch_size)
-                for j, p in zip(idx, liveness_prob):
-                    video_id = dataset.get_video_id(j)
-                    df.iloc[j] = {'prob': p.item(), 'video_id': video_id}
+        for mode in ['dev', 'test']:
+            dataset = CasiaSurfDataset(protocol, mode=mode, transform=transforms.Compose([
+                transforms.Resize((320, 240)),
+                transforms.ToTensor()
+            ]))
+            dataloader = data.DataLoader(dataset, batch_size=args.batch_size)
```
Comment (grib0ed0v) [inline submit.py]: How about adding num_workers to DataLoader?

File: test.py [hunk 3]
```diff
@@ -43,13 +42,13 @@ def evaluate(dataloader: data.DataLoader, model: nn.Module):
     argparser.add_argument('--num_classes', type=int, default=2)
     argparser.add_argument('--batch_size', type=int, default=1)
     args = argparser.parse_args()
-    dataset = CasiaSurfDataset(args.protocol, dir=args.data_dir, transform=transforms.Compose([
+    dataset = CasiaSurfDataset(args.protocol, mode='dev', dir=args.data_dir, transform=transforms.Compose([
         transforms.Resize((320, 240)),
         transforms.ToTensor()
     ]))
-    dataloader = SplittedDataLoader(dataset, train_batch_size=1, val_batch_size=args.batch_size)
+    dataloader = data.DataLoader(dataset, batch_size=args.batch_size)
```
Comment (grib0ed0v) [inline test.py]: num_workers?

**Review by grib0ed0v — CHANGES_REQUESTED** [review]

**Pull request merged**